gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -330,6 +330,78 @@ gst_tensor_filter_destroy_notify (void *data)
   }
 }
 
+/**
+ * @brief Prepare statistics for performance profiling (e.g, latency, throughput)
+ */
+static void
+prepare_statistics (GstTensorFilterPrivate * priv)
+{
+  priv->stat.latest_invoke_time = g_get_real_time ();
+}
+
+/**
+ * @brief Helper function to accumulate latencies
+ */
+static void
+accumulate_latency (void *data, void *user_data)
+{
+  gint64 *latency = data;
+  gint64 *total_latency = user_data;
+
+  *total_latency += *latency;
+}
+
+/**
+ * @brief Record statistics for performance profiling (e.g, latency, throughput)
+ */
+static void
+record_statistics (GstTensorFilterPrivate * priv)
+{
+  gint64 end_time = g_get_real_time ();
+  gint64 *latency = g_new (gint64, 1);
+  GQueue *recent_latencies = priv->stat.recent_latencies;
+
+  *latency = end_time - priv->stat.latest_invoke_time;
+  priv->stat.total_invoke_latency += *latency;
+  priv->stat.total_invoke_num += 1;
+
+  if (g_queue_get_length (recent_latencies) == GST_TF_STAT_MAX_RECENT)
+    g_free (g_queue_pop_head (recent_latencies));
+  g_queue_push_tail (recent_latencies, latency);
+
+  /* the queue should have at least one element */
+  g_assert (g_queue_get_length (recent_latencies) != 0);
+
+  if (priv->latency_mode > 0) {
+    gint64 avg_latency;
+
+    g_queue_foreach (recent_latencies, accumulate_latency, &avg_latency);
+    avg_latency /= g_queue_get_length (recent_latencies);
+
+    if (avg_latency > INT32_MAX)
+      priv->prop.latency = -1;
+    else
+      priv->prop.latency = (gint) avg_latency;
+  }
+
+  if (priv->throughput_mode > 0) {
+    gint throughput_int = -1;
+
+    if (priv->stat.total_invoke_latency != 0) {
+      gdouble throughput =
+          (gdouble) (priv->stat.total_invoke_num * G_USEC_PER_SEC);
+
+      throughput /= priv->stat.total_invoke_latency;
+      throughput *= 1000;
```

Comment:
For the sake of accuracy, do:
```
throughput = (num * USEC_PER_SEC * 1000.0) / total_latency;
```

Reply:
Plus, throughput-related variables are monotonously increasing; thus you are averaging the total execution, which worries me with overflows.

Reply:
BUG: if latency = 0 and througput = 1, you may do divide-by-zero.

Reply:
> For the sake of accuracy, do: throughput = (num * USEC_PER_SEC * 1000.0) / total_latency;

Updated, thanks.

> Plus, throughput-related variables are monotonously increasing; thus you are averaging the total execution, which worries me with overflows.

You're right! I changed codes to cache old values if the value exceeds some threshold. But, I'm not sure it's a good way.

> BUG: if latency = 0 and througput = 1, you may do divide-by-zero. and total_invoke_latency is incorrect.

Hmm. I think I don't understand your concerns. Even if `latency_mode = 0` and `throuput_mode = 1`, `total_invoke_latency` is updated. And... the if-statement may prevent divide-by-zero.